ecl/hql/hqllex.l

```diff
@@ -64,6 +64,94 @@ extern void hex2str(char * target, const char * digits, unsigned len);
 #define RETURNHARD(sym) \
         setupdatepos; return sym;
 
+//#define TEST_ID_SUGGESTIONS
+#ifdef TEST_ID_SUGGESTIONS
+static char getNewIdChar(unsigned index)
+{
+    //1st char of an ID must be an alpha [a-z]
+    if (index == 0)
+        return 'a' + rand()%26;
+
+    unsigned n = rand()%37;
+    if (n < 10)
+        return '0' + n;
+    else if(n == 10)
+        return '_';
+    else if (n > 10  && n < 37)
+        return 'a' + n - 11;
+    else
+        throwUnexpected();
+}
+
+static unsigned scrambledIdCount = 0;
+
+class DummyAtom : public CInterfaceOf<IIdAtom>
+{
+public :
+    DummyAtom(const void * k): hash(0)
+    {
+        text = strdup((const char *)k);
+        lower = new LowerCaseAtom(text);
+    }
+    ~DummyAtom()
+    {
+        free(text);
+        lower->Release();
+    }
+
+//interface:IMapping
+    virtual const char *        queryStr() const { return text; }
+    virtual const void *        getKey() const { return text; }
+    virtual unsigned            getHash() const { return hash; }
+    virtual void                setHash(unsigned _hash) { hash = _hash; }
+    virtual IAtom *             queryLower() const { return lower; }
+
+private:
+    unsigned hash;
+    char * text;
+    IAtom * lower;
+};
+
+static IIdAtom * createScrambledIdAtom(const IIdAtom * idToScramble, unsigned nCharToScramble, unsigned probability, unsigned maxIdsToScramble)
```

Comment:
This should be in a different file - e.g., hqlparse.cpp.  It is confusing the common case code with some debug test code.